### DISCUSSION TITLE
Why does libultrahdr use P3 gamut and rec601 rgb2yuv matrix for API-0?

### DISCUSSION THREAD

**Discussion by zhengch01** [google/libultrahdr#163] [Q&A] [answered]
I'm just curious why did the Encode API-0 use P3 gamut and rec601 rgb2yuv matrix for processing? How does other jpeg decoder library or renderer know which yuv2rgb matrix should be applied for display? I think there should be a problem for non-libultrahdr decoder. In my opinion, it's more reasonable to use sRGB in the whole workflow.

**Comment by DichenZhang1** [comment, marked as answer]
Hi @zhengch01,

Thank you for your questions.

Re "why did the Encode API-0 use P3 gamut and rec601 rgb2yuv matrix for processing?"
This was indeed a bug and is fixed in the latest version.

Re "How does other jpeg decoder library or renderer know which yuv2rgb matrix should be applied for display?"
They should read the ICC package which has the matrix.

Please let me know if there is more questions.

Thank you!

**Reply by zhengch01** [reply]
Thanks for your explanation. :-)

**Reply by ValeZAA** [reply]
>opinion, it's more reasonable to use sRGB in the whole workflow.

Please note that techincally sRGB is a transfer function and can be used with P3-D65 primaries and Bt.601 full range matrix, matrix typical for JPEG.
